nx/lib/nx.ex

```diff
@@ -6612,6 +6612,152 @@ defmodule Nx do
   defp mean_den(shape, [axis | axes]) when axis >= 0,
     do: elem(shape, axis) * mean_den(shape, axes)
 
+  @doc """
+  Returns the weighted mean for the tensor and the weights.
+
+  If the `:axis` option is given, it aggregates over
+  that dimension, effectively removing it. `axes: [0]`
+  implies aggregating over the highest order dimension
+  and so forth. If the axis is negative, then counts
+  the axis from the back. For example, `axis: [-1]` will
+  always aggregate all rows. You can only aggregate over
+  one axis.
```

Comment:
```suggestion
  always aggregate over the last dimension.
```

We shouldn't use rows for N-dim tensors. Also, "axis" being singular is our standard for "single-axis aggregation"